### ISSUE TITLE
snoopd not reading POST codes from snoop device

### ISSUE THREAD

**Issue by zevweiss** [openbmc/phosphor-host-postd#1]
After updating my development tree from a newer upstream commit, `snooper` stopped producing POST codes on my system.  I traced this back to `snoopd` not reading the codes from the snoop device, which appears to be because immediately after adding the snoop device file descriptor to its epoll wait set, it immediately (and erroneously) removes it:

```
root@asrock-e3c246d4i:~# ./strace -e openat,close,epoll_create1,epoll_ctl,epoll_wait -f /usr/bin/snoopd -v -b 1 -d /dev/aspeed-lpc-snoop0
...
openat(AT_FDCWD, "/dev/aspeed-lpc-snoop0", O_RDONLY|O_NONBLOCK|O_LARGEFILE) = 3
...
epoll_create1(EPOLL_CLOEXEC)            = 5
epoll_ctl(5, EPOLL_CTL_ADD, 3, {EPOLLIN|EPOLLET, {u32=31919216, u64=31919216}}) = 0
epoll_ctl(5, EPOLL_CTL_DEL, 3, NULL)    = 0
...
```

While I'm not 100% certain of this, my guess is that this is due to a change in commit aade4adb3987778a3e6f8e2e623109563cf7fd72 moving `reporterSource` into an `if` block, and the `EPOLL_CTL_DEL` is happening due to the object's destructor getting run at the end of that block, before the main event loop starts (whereas previously that object's lifetime lasted through the call to `event.loop()`).

**Comment by zevweiss** [comment]
Ah, and after checking gerrit I see this has already had a patch submitted to fix it: https://gerrit.openbmc-project.xyz/c/openbmc/phosphor-host-postd/+/41787

**Comment by zevweiss** [comment]
Fixed by commit 2a744b2d70ce9de8519a7c716da5009cb049db17.
